src/internal/toHexPrice.ts

```diff
@@ -0,0 +1,20 @@
+import { BigIntish } from '@brinkninja/types'
+
+const DECIMALS = 18;
+
+export function convertToHexPrice (tokenOutAmount: BigIntish, tokenInAmount: BigIntish): string {
+  const bigTokenOutAmount = convertToBigInt(tokenOutAmount);
+  const bigTokenInAmount = convertToBigInt(tokenInAmount);
+
+  const computedPrice = bigTokenOutAmount * (2n ** 96n) / bigTokenInAmount;
+  const hexPrice = '0x' + computedPrice.toString(16);
+  return hexPrice;
+}
+
+const convertToBigInt = (value: BigIntish): bigint => {
```

Comment:
i feel using BigIntish internally to be a nightmare (and union types in general). string, bigint and number expose completely different interfaces.

Reply:
> accepting are integers (like 1234), integers as strings (like "1234")

But what if they need decimals? e.g. i want to swap 0.1 ETH for 1 DAI.

Reply:
misunderstood. addressed